### PR TITLE
feat: add `@joshdb/maria`

### PR DESCRIPTION
**Pull request by j122j**

### CODE REVIEW

File: packages/mariadb/rollup.config.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      sourcemap: true
+    },
+    {
+      file: './dist/index.mjs',
+      format: 'es',
+      exports: 'named',
+      sourcemap: true
+    }
+  ],
+  external: [],
```
Comment (dan-online) [inline packages/mariadb/rollup.config.ts:22]: This needs to list external dependencies, check mongoose for an example or check the warnings when you build this package

File: packages/mariadb/src/lib/MariaDBProvider.ts [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+  Method,
+  Payloads
+} from '@joshdb/provider';
+import { Snowflake, TwitterSnowflake } from '@sapphire/snowflake';
+import { isPrimitive } from '@sapphire/utilities';
+import type { ConnectionConfig as MariaDBConnectionConfig } from 'mariadb';
+import { deleteProperty, getProperty, hasProperty, PROPERTY_NOT_FOUND, setProperty } from 'property-helpers';
+import { QueryHandler } from './QueryHandler';
+
+export class MariaDBProvider<StoredValue = unknown> extends JoshProvider<StoredValue> {
```
Comment (dan-online) [inline packages/mariadb/src/lib/MariaDBProvider.ts:29]: Perhaps we just call it "MariaProvider"

**Review by dan-online — CHANGES_REQUESTED** [review]
Just a comment or two, I don't see anything completely wrong based on my limited knowledge of SQL, I do wonder though, how secure the queries are? Is some extra Santi action required or is the method you used the best practice?

**Comment by j122j** [comment]
> Just a comment or two, I don't see anything completely wrong based on my limited knowledge of SQL, I do wonder though, how secure the queries are? Is some extra Santi action required or is the method you used the best practice?

Keys and values sanitized using placeholders. I think there could be an SQL injection with tableName which is provided by the developer.

File: packages/mariadb/src/lib/QueryHandler.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
+import { toJSON, toRaw } from '@joshdb/serialize';
+import mariadb from 'mariadb';
```
Comment (RealShadowNova) [inline packages/mariadb/src/lib/QueryHandler.ts:2]: Should be using named exports wherever possible

**Review by RealShadowNova — CHANGES_REQUESTED** [review]

**Comment by dan-online** [comment]
> Keys and values sanitized using placeholders. I think there could be an SQL injection with tableName which is provided by the developer.

Yea alright cool

**Comment by RealShadowNova** [comment]
Blocked by #128

**Review by RealShadowNova — APPROVED** [review]

**Review by dan-online — APPROVED** [review]
LGTM

File: .github/workflows/continuous-tests.yml [hunk 4]
```diff
@@ -50,6 +50,33 @@ jobs:
...
+    name: MariaProvider
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3
+      - name: Use Node.js v16
+        uses: actions/setup-node@2fddd8803e2f5c9604345a0b591c3020ee971a93 # tag=v3
+        with:
+          node-version: 16
+      - name: Shutdown MySQL
```
Comment (RealShadowNova) [inline .github/workflows/continuous-tests.yml:63]: Why is the mysql database being shutdown and then started?

**Review by RealShadowNova — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous-tests.yml (hunk 4 above)
Comment (dan-online) [reply .github/workflows/continuous-tests.yml:63]: It's the recommended method in the Maria documentation, afaik GitHub actions have an already running MySQL server that needs to be stopped before starting a mariadb

**Review by RealShadowNova — APPROVED** [review]

**Pull request merged**